contracts/nft/src/core/msg.rs

```diff
@@ -0,0 +1,209 @@
+use cosmwasm_schema::{cw_serde, QueryResponses};
+use cosmwasm_std::{Addr, Binary};
+use cw2::ContractVersion;
+use cw_ownable::{cw_ownable_execute, cw_ownable_query};
+use cw_utils::Expiration;
+use provwasm_std::types::provenance::metadata::v1::ScopeResponse;
+
+#[cw_serde]
+pub enum InstantiateMsg {
+    Default {
+        /// The only address which can create new NFTs
+        minter: Addr,
+        /// Name of the NFT contract
+        name: String,
+        /// Symbol of the NFT contract
+        symbol: String,
+        /// Generated UUID to use for creating the Contract Specification
+        contract_spec_uuid: String,
+        /// Generated UUID to use for creating the Scope Specification
+        scope_spec_uuid: String,
+    },
+}
+
+/// CW721 execute methods in addition to `Mint`, `Burn`, and `Change Ownership`
+#[cw_ownable_execute]
+#[cw_serde]
+pub enum ExecuteMsg {
+    /// Allow operator to transfer / send the token from the owner's account.
+    /// If expiration is set, then this allowance has a time/height limit
+    Approve {
+        spender: Addr,
+        /// Token ID in UUID format (Scope UUID)
+        token_id: String,
+        expires: Option<Expiration>,
+    },
+
+    /// Allows operator to transfer / send any token from the owner's account.
+    /// If expiration is set, then this allowance has a time/height limit
+    ApproveAll {
+        operator: Addr,
+        expires: Option<Expiration>,
+    },
+
+    /// Burn an NFT the sender has access to
+    Burn {
+        /// Token ID in UUID format (Scope UUID)
+        token_id: String,
+    },
+
+    /// Mint a new NFT, can only be called by the contract minter
+    Mint {
+        /// Generated UUID to use for creating the NFT (Scope)
+        scope_uuid: String,
+        /// Generated UUID to use for creating the Session which encapsulates an owner data record
+        session_uuid: String,
+        /// Recipient of the NFT
+        recipient: Addr,
+    },
+
+    /// Remove previously granted Approval
+    Revoke {
+        spender: Addr,
+        /// Token ID in UUID format (Scope UUID)
+        token_id: String,
+    },
+
+    /// Remove previously granted ApproveAll permission
+    RevokeAll { operator: Addr },
+
+    /// Transfer a token to a contract and trigger an action on the receiving contract.
+    SendNft {
+        /// Generated `UUID` to use for creating the `NFT`/`Scope`
+        token_id: String,
+        /// Generated `UUID` to use for creating the `Session` to write the new owner record
+        session_uuid: String,
+        /// Receiving Contract Address
+        contract: Addr,
+        /// Message to send to receiving contract
+        msg: Binary,
+    },
+
+    /// Transfer a token to another account without triggering actions
+    TransferNft {
+        /// Generated `UUID` to use for creating the `NFT`/`Scope`
+        token_id: String,
+        recipient: Addr,
```

Comment:
Very minor, but noticed `recipient` doesn't have a comment.

Reply:
I originally thought the main message covered, but I added since more == better